Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -526,12 +526,10 @@ class BrowserViewController: UIViewController {
     }
 
     private func removeBookmark(url: String) {
-        profile.bookmarks.removeByURL(url, success: { success in
-            self.toolbar?.updateBookmarkStatus(!success)
-            self.urlBar.updateBookmarkStatus(!success)
-        }, failure: { err in
-            log.error("Error removing bookmark: \(err).")
-        })
+        profile.bookmarks.removeByURL(url) >>== {
+            self.toolbar?.updateBookmarkStatus(false)
+            self.urlBar.updateBookmarkStatus(false)
```

Comment:
Careful. This will be run on the background queue, and these look like UI ops to me. You might want to do the same thing as in `BookmarksPanel`:

```
.removeByURL(url).uponQueue(dispatch_get_main_queue()) { res in
```